### PR TITLE
When calling compileAsync(), fallback to renderAsync

### PR DESCRIPTION
**Pull request by RobLoach**
When calling compileAsync(), we should fallback to renderAsync(). Not sure if this is done right.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -159,6 +159,12 @@ Transformer.prototype.compile = function (str, options) {
 };
 Transformer.prototype.compileAsync = function (str, options, cb) {
   if (!this.can('compileAsync')) { // compileFile* || renderFile* || renderAsync || compile*Client*
+    if (this.can('renderAsync')) {
+      var customCompile = function (str, options) {
+        return this.renderAsync(str, options);
+      }
+      return tr.normalizeFnAsync(customCompile, cb);
+    }
     return Promise.reject(new Error('The Transform "' + this.name + '" does not support compiling plain strings')).nodeify(cb);
```
Comment (RyanZim) [inline index.js]: This should be the only statement in this conditional.

**Comment by RyanZim** [comment]
> Not sure if this is done right.

`renderAsync` should be added to the list of fallbacks  for `compileAsync` above.

The error should be the only statement inside `if (!this.can('compileAsync'))`.

After `if (this._hasMethod('compileAsync'))`, you should add an `else if (this.can('renderAsync'))` block, containing your call to `renderAsync`.

Hope this is clear enough.

**Comment by RobLoach** [comment]
@RyanZim Still not working. Added a test. Know what it's missing? Feel free to commit directly to the branch.

**Comment by RyanZim** [comment]
> When calling compileAsync(), we should fallback to renderAsync()

We can't. `compileAsync()` has to return `{fn: Function, dependencies: Array.<String>}` and `fn` has to return a string. If we use `renderAsync()` under the hood, `fn` will return a promise, not a string.

I never thought of this before, but since there's no way to make something async return synchronously, we're out of options. It just won't work.

I guess close this. Is there a particular transformer that you need this for?

**Comment by RyanZim** [comment]
@RobLoach ping?

**Comment by RobLoach** [comment]
You're right, we can't do it unfortunately. Maybe best way to do it would be implementing both functions in the transformers... I'll close this for now. Thanks!

**Pull request closed without merging**